docs/specs/authentication-flow-selection.md

```diff
@@ -0,0 +1,156 @@
+
+For example, a project can have multiple login flow and signup flows. After defining the flow groups and allowlist, the client app can specify which flow group to use using `x_authentication_flow_group` in the [authentication request](/docs/specs/oidc.md#x_authentication_flow_group).
+
+### Flow Allowlist
+
+Flow group only instructs default UI to use a specific group of flows. To allow them to be used in custom UI, the flow group must be defined in the allowlist configuration.
+
+## Flow Selection in Different UIs
+
+There are two types of UIs:
+- **Default UI**: The default UI provided by Authgear.
+- **Custom UI**: A custom UI that uses Authgear's Authflow API to initiate a flow.
+
+They have different behaviours when selecting authentication flows.
+
+### Default UI
+
+Default UI uses flow group for deriving allowed flows.
+
+Following table explains how authentication flow is selected in the default UI:
+
+| Flow allowlist | Authorization Request | Behaviour |
+| --- | --- | --- |
+| Defined | `x_authentication_flow_group` is present | Selected flow group is checked against the allowlist and used if valid. |
+| Defined | `x_authentication_flow_group` is **NOT** present | First flow group in the allowlist is used. |
+| Not defined | `x_authentication_flow_group` is present | Selected flow group is used. |
+| Not defined | `x_authentication_flow_group` is **NOT** present | Default flow group is used. |
+
+### Custom UI
+Unlike Default UI, Custom UI creates flows using Authflow API. `x_authentication_flow_allowlist` is not used to restrict the flows created by Custom UI.
+
+Instead, it computes the effective flow allowlist using union of group and individual flows in the allowlist to decide which flows can be created.
+
+## Configuration
+
+Each client app may have an allowlist of flow groups defined with `x_authentication_flow_allowlist`. Once specified, the authorization request will be rejected if specified flow group is not in the allowlist.
+
+For example, given following flow groups defined:
+
+```yaml
+ui:
+  authentication_flow:
+    groups:
+    - name: oauth_only
+      login_flows:
+      - oauth_only
+      signup_flows:
+      - oauth_only
+      - email_password_2fa
+      signup_login_flows:
+      - oauth_only
+      reauth_flows:
+      - password
+      promote_flow:
+      - password
+      account_recovery_flows:
+      - email
+    - name: email_password_2fa
+      login_flows:
+      - email_password_2fa
+      signup_flow:
+      - email_password_2fa
+      signup_login_flow:
+      - email_password_2fa
+      reauth_flow:
+      - sms_code
+      promote_flow:
+      - sms_code
+      account_recovery_flow:
+      - sms
+```
+
+You can specify allowed flow groups and individual flows for each client app:
+
+```yaml
+oauth:
+  clients:
+  - client_id: public_app
+    x_authentication_flow_allowlist:
+      # Only allow OAuth login for public_app
+      groups:
+      - oauth_only
```

Comment:
Rename to `- name: oauth_only`. That is, change from a plain string to a struct with a field called `name`.